dashboard/app/controllers/registrations_controller.rb

```diff
@@ -154,10 +211,12 @@ def create
       super
     end
 
-    if current_user && current_user.errors.blank?
-      if current_user.teacher?
+    curr_user = !!params[:new_sign_up] ? User.find_by_email(params[:user_email]) : current_user
```

Comment:
Oh I was noticing that by that point, the user was considered signed-in (and would have `current_user` set) for the old sign-up flow but would not for the new one, so I was ensuring a user was found for the remaining logic. But actually I think it may make more sense to ensure the user is signed in here then just use `current_user` like you said!